lambdas/mail_save.js

```diff
@@ -37,6 +40,24 @@ const handler = async (req, res) => {
               statuses['created'],
               'La suscripción se ha realizado correctamente'
             );
+
+            fs.readFile(path.resolve(__dirname, '../templates/mail_registration.html'), 'utf8', function(err, html) {
```

Comment:
1. Why the `function`? You are using arrow everywhere.
2. I'm not entirely sure how the lifecycle of a lambda function works; but we should probably keep de file contents on a variable and add a _"private function"_ to load it if needed.

Reply:
I suggest using async/await (or in this case, fs.readFileSync) to prevent unneeded function stacking.

@homer0 I don't see any particular need for what you're describing in 2), and I can't see the relation with the lambda lifecycle. Care to clarify a bit further?

Reply:
@joelalejandro By lifecycle I mean that I don't know if the middleware/function is mounted the moment we deploy and it stays _"alive"_ all the time, of it works like the _"free heroku deployments"_, that are suspended until it gets a request (like the BeerJS's slack invitation page).

If the function is _"alive"_ all the time, reading the html template from the filesystem every time it has to handle a request seems unnecessary, you could just load it once and prevent further attempts if the initial load failed:

```js
let emailTemplate;
let emailTemplateError = false;
const getEmailTemplate() {
  if (emailTemplateError) {
    return;
  }
  if (!emailTemplate) {
    try {
      emailTemplate = fs.readFileSync(
        path.resolve(__dirname, '../templates/mail_registration.html'),
        'utf8'
      );
    } catch (error) {
      ...do something with the error.
      emailTemplateError = true;
    }
  }

  return emailTemplate;
}

...
const html = getEmailTemplate();
if (html) {
  ...
}
```

Reply:
AFAIK, each function implemented is stateless, so we can't store global variables to optimize runs by reading just once the template file.

Reply:
@spersico I got the idea from here: [Stateless doesn't mean no state](https://www.trek10.com/blog/stateless/) (Yes, the title, I know).

Reply:
umm... okay, we don't lose anything with your implementation, really. Will try it out, I guess we'll see when it's deployed